### ISSUE TITLE
Useless getter in Application class

### ISSUE THREAD

**Issue by LJNeon** [TypeStrong/typedoc#1235]
- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Unexpected Behavior
[Here](https://github.com/TypeStrong/typedoc/blob/master/src/lib/application.ts#L149-L154) there's a getter that simply calls `return this;`. I can't imagine any use cases for this.

## Environment
  - Typedoc version: GitHub Master

**Comment by Gerrit0** [comment]
TypeDoc has a "component" architecture, most of the old classes in the project extend the [`AbstractComponent`](https://github.com/TypeStrong/typedoc/blob/master/src/lib/utils/component.ts#L105) class.

This class provides a getter - `application` which is used for any classes in the component hierarchy to access the root level component. Once we get to the `Application` class though, we don't want to bubble up through the component owners anymore, and just want to return the current object.

Technically, the getter is no longer necessary. Before #845, the `AbstractComponent` class would return `null` if the component had no owner and therefore the `Application` class needed to override the method. Now, there is a check to satisfy strict null checks (with a type assertion that needs to go away eventually).

I want to remove the `@Component` decorator and the hard to follow component class structure along with it, but this needs to be done in parts to keep the changes small enough that they probably won't break things. (It has already been done to the options + options readers, next up is probably the converters)
